packages/x-data-grid/src/models/props/DataGridProps.ts

```diff
@@ -417,6 +417,10 @@ export interface DataGridPropsWithoutDefaultValue<R extends GridValidRowModel =
    * Override the height/width of the Data Grid inner scrollbar.
    */
   scrollbarSize?: number;
+  /**
+   * Override the height/width of the header filters.
+   */
+  filterHeaderHeight?: number;
```

Comment:
Also, shouldn't this prop belong to `dataGridProProps.ts` like [`headerFilters`](https://github.com/mui/mui-x/blob/ddd0f13afa4bea7c5bbd12b399f370cc6bc286e4/packages/x-data-grid-pro/src/models/dataGridProProps.ts#L139) boolean flag?

Reply:
I need it in community for the dimensions logic.

Reply:
What was the reason for splitting `DataGridProcessedProps` and `DataGridProcessedPropsWithShared`? Having some trouble fixing the types because `DataGridProcessedProps` is used in many places where `DataGridProcessedPropsWithShared` should be used. AFAIK, `DataGridProcessedProps` is not used by external users so we should be able to place final typings there.

Reply:
Did the refactor. Feels better this way because the typings reflect what's really present.